pramen-py/src/pramen_py/runner/runner_transformation.py

```diff
@@ -67,16 +69,25 @@ def overwrite_info_dates(
     return config
 
 
-def discover_transformations(
-    path: pathlib.Path = TRANSFORMATIONS_DIR,
-) -> Generator[Type[T_TRANSFORMATION], None, None,]:
+def get_transformations_dir(dir_key: str, envs: Env) -> pathlib.Path:
+    transformations_dir_path = os.environ.get(dir_key) or ""
+    if not transformations_dir_path:
+        transformations_dir_path = envs.str(dir_key, "")
```

Comment:
there is no need to check for env var in the environment and then read env.str.... env.read_env (we have it in the beginning) loads into the main process environment variables from the .env file. In case of conflict of the name, the one from the environment takes precedence. Then once you access env.str it returns you the value of env var from the enivornment.